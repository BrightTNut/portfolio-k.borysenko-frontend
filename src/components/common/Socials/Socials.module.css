.container {
  position: absolute;
  top: 50%;
  right: -180px;
  transform: translateY(-50%);
  padding: 40px 30px;
  box-sizing: border-box;
  background-color: var(--bg-primary);
  box-shadow: rgba(0, 0, 0, 0.24) 0px 3px 8px;
}

.title {
  font-family: var(--font-heading);
  margin-bottom: 30px;
}

.item {
  display: flex;
  align-items: center;
}

.item:not(:last-child) {
  margin-bottom: 8px;
}

.item:nth-child(2) {
  margin-bottom: 12px;
}

.item:nth-child(3) {
  margin-bottom: 10px;
}

.name {
  font-size: var(--font-size-caption);
  margin-left: 18px;
  letter-spacing: 1px;
  transition: color 250ms cubic-bezier(0.4, 0, 0.2, 1);
}

.name:hover {
  color: var(--ui-hover);
}
