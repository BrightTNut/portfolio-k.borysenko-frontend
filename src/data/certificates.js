import frontendInfopulseCertificate from "../images/certificates/infopulse.jpg";
import englishPrimeCertificate from "../images/certificates/english.jpeg";
import fullstackGoItCertificate from "../images/certificates/fullstack.jpeg";
import designAlpacaSchoolCertificate from "../images/certificates/design.png";

import { INFOPULSE_UNIVER_URL, ENGLISH_PRIME_SCHOOL_URL, GO_IT_URL, ALPACA_SCHOOL_URL } from '../utils/urls'


export const certificates = [
  {
    id: 1,
    imgSrc: frontendInfopulseCertificate,
    alt: 'Frontend Developer Certificate',
    width: '736px',
    title: 'Infopulse Univer',
    url: INFOPULSE_UNIVER_URL,
    city: 'Kyiv',
    year: '2020',
    duration: '3 months'
  },
  {
    id: 2,
    imgSrc: englishPrimeCertificate,
    alt: 'English  Certificate',
    width: '736px',
    title: 'English Prime School',
    url: ENGLISH_PRIME_SCHOOL_URL,
    city: 'Kyiv',
    year: '2021',
    duration: '18 months'
  },
  {
    id: 3,
    imgSrc: fullstackGoItCertificate,
    alt: 'Fullstack Developer Certificate',
    width: '520px',
    title: 'Go It Bootcamp',
    url: GO_IT_URL,
    city: 'Kyiv',
    year: '2022',
    duration: '8 months'

  },
  {
    id: 4,
    imgSrc: designAlpacaSchoolCertificate,
    alt: 'Designer Certificate',
    width: '736px',
    title: 'Alpaca School',
    url: ALPACA_SCHOOL_URL,
    city: 'Kyiv',
    year: '2023',
    duration: '1 week'
  },
]

