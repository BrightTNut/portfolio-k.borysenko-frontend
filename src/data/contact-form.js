import { UserIcon, EmailIcon, PhoneIcon, ClockIcon, MessageIcon } from "../components/icons";

export const inputConfig = [{
  id: 1,
  name: 'name',
  label: 'YOUR NAME',
  icon: <UserIcon />,
  // }, {
  //   id: 2,
  //   name: 'email',
  //   label: 'E-MAIL',
  //   icon: <EmailIcon />,
  // }, {
  //   id: 3,
  //   name: 'phone',
  //   label: 'PHONE NUMBER (optional)',
  //   icon: <PhoneIcon />,
  // }, {
  //   id: 4,
  //   name: 'time',
  //   label: 'PREFERRED TIME OF CALL BACK',
  //   icon: <ClockIcon />,
  // }, {
  //   id: 5,
  //   name: 'message',
  //   type: 'textarea',
  //   label: 'MESSAGE',
  //   icon: <MessageIcon />,
},]