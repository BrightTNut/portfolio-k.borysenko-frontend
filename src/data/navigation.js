export const navConfig = [{
  id: 1,
  name: "Home",
  path: "/",
}, {
  id: 2,
  name: "About Me",
  path: "/about",
}, {
  id: 3,
  name: "Skills",
  path: "/skills",
}, {
  id: 4,
  name: "Projects",
  path: "/projects",
}, {
  id: 5,
  name: "Certificates",
  path: "/certificates",
},
{
  id: 6,
  name: "Feedbacks",
  path: "/feedbacks",
},
{
  id: 7,
  name: "Contacts",
  path: "/contacts",
},]