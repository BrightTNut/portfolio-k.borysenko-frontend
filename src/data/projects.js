import univer from '../images/projects/univer.png';
import portfolio from '../images/projects/portfolio.png';
import todo from '../images/projects/todo.png';
import filmoteka from '../images/projects/univer.png';

export const projects = [
  {
    id: 1,
    title: 'Univer React App',
    path: univer,
    desc: 'Lorem ipsum dolor sit amet consectetur adipisicing elit. Dolores ab id ad nesciunt quo aut corporis modi? Voluptate, quos sunt dolorum facilis, id eum sequi placeat accusantium saepe eos laborum.',
    alt: 'Univer Logo',
    techStack: ['React', 'Redux', 'Type Script'],
    codeLink: '',
    demoLink: ''

  },
  {
    id: 2,
    title: 'Portfolio React App',
    path: portfolio,
    desc: 'Lorem ipsum dolor sit amet consectetur adipisicing elit. Dolores ab id ad nesciunt quo aut corporis modi? Voluptate, quos sunt dolorum facilis, id eum sequi placeat accusantium saepe eos laborum.',
    alt: 'Portfolio Logo',
    techStack: ['React', 'Redux', 'Type Script', 'React Testing Library'],
    codeLink: '',
    demoLink: ''
  },
  {
    id: 3,
    title: 'Todo React App',
    path: todo,
    desc: 'Lorem ipsum dolor sit amet consectetur adipisicing elit. Dolores ab id ad nesciunt quo aut corporis modi? Voluptate, quos sunt dolorum facilis, id eum sequi placeat accusantium saepe eos laborum.',
    alt: 'Todo Logo',
    techStack: ['React', 'Redux', 'Type Script', 'React Testing Library'],
    codeLink: '',
    demoLink: ''
  },

];