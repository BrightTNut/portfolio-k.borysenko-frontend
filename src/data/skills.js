export const skills = [
  {
    id: 1,
    name: 'Agile',
    path: '../images/skills/agile.png',
    type: 'Development Tools'
  },
  {
    id: 2,
    name: 'Apollo',
    path: '../images/skills/apollo.png',
    type: 'Frameworks and Libraries'
  },
  {
    id: 3,
    name: 'CSS3',
    path: '../images/skills/css3.png',
    type: 'Programming Languages'
  },
  {
    id: 4,
    name: 'Express',
    path: '../images/skills/express.png',
    type: 'Frameworks and Libraries'
  },
  {
    id: 5,
    name: 'Figma',
    path: '../images/skills/figma.png',
    type: 'Testing and Design'
  },
  {
    id: 6,
    name: 'Firebase',
    path: '../images/skills/firebase.png',
    type: 'Databases and APIs'
  },
  {
    id: 7,
    name: 'Formik',
    path: '../images/skills/formik.png',
    type: 'Frameworks and Libraries'
  },
  {
    id: 8,
    name: 'Git',
    path: '../images/skills/git.png',
    type: 'Development Tools'
  },
  {
    id: 9,
    name: 'GitHub',
    path: '../images/skills/github.png',
    type: 'Development Tools'
  },
  {
    id: 10,
    name: 'GitLab',
    path: '../images/skills/gitlab.png',
    type: 'Development Tools'
  },
  {
    id: 11,
    name: 'GraphQL',
    path: '../images/skills/graphql.png',
    type: 'Databases and APIs'
  },
  {
    id: 12,
    name: 'HTML5',
    path: '../images/skills/html5.png',
    type: 'Programming Languages'
  },
  {
    id: 13,
    name: 'Jest',
    path: '../images/skills/jest.png',
    type: 'Testing and Design'
  },
  {
    id: 14,
    name: 'Jira',
    path: '../images/skills/jira.png',
    type: 'Development Tools'
  },
  {
    id: 15,
    name: 'JS',
    path: '../images/skills/js.png',
    type: 'Programming Languages'
  },
  {
    id: 16,
    name: 'Next JS',
    path: '../images/skills/next-js.png',
    type: 'Frameworks and Libraries'
  },
  {
    id: 17,
    name: 'Node JS + Express',
    path: '../images/skills/node-js-express.jpeg',
    type: 'Frameworks and Libraries'
  },
  {
    id: 18,
    name: 'React Hook Form',
    path: '../images/skills/react-hook-form.png',
    type: 'Frameworks and Libraries'
  },
  {
    id: 19,
    name: 'React Native',
    path: '../images/skills/react-native.png',
    type: 'Frameworks and Libraries'
  },
  {
    id: 20,
    name: 'React Testing Library',
    path: '../images/skills/react-testing-library.png',
    type: 'Testing and Design'
  },
  {
    id: 21,
    name: 'React',
    path: '../images/skills/react.png',
    type: 'Frameworks and Libraries'
  },
  {
    id: 22,
    name: 'Redux',
    path: '../images/skills/redux.png',
    type: 'Frameworks and Libraries'
  },
  {
    id: 23,
    name: 'REST API',
    path: '../images/skills/rest-api.png',
    type: 'Databases and APIs'
  },
  {
    id: 24,
    name: 'SASS',
    path: '../images/skills/sass.png',
    type: 'Testing and Design'
  },
  {
    id: 25,
    name: 'Styled components',
    path: '../images/skills/styled-components.png',
    type: 'Testing and Design'
  },
  {
    id: 26,
    name: 'Trello',
    path: '../images/skills/trello.png',
    type: 'Development Tools'
  },
  {
    id: 27,
    name: 'Mongo DB',
    path: '../images/skills/mongodb.png',
    type: 'Databases and APIs'
  },
]
