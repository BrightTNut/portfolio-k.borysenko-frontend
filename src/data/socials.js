import { Facebook, GithubIcon, InstagramIcon, LinkedinIcon } from '../components/icons'
import { GITHUB_URL, LINKEDIN_URL, INSTAGRAM_URL, FACEBOOK_URL } from '../utils/urls'

export const socials = [{
  id: 1,
  name: 'GIT HUB',
  icon: <GithubIcon width='44px' />,
  url: GITHUB_URL
}, {
  id: 2,
  name: 'LINKED IN',
  icon: <LinkedinIcon width='44px' />,
  url: LINKEDIN_URL
}, {
  id: 3,
  name: 'INSTAGRAM',
  icon: <InstagramIcon width='44px' />,
  url: INSTAGRAM_URL
}, {
  id: 4,
  name: 'FACEBOOK',
  icon: <Facebook width='44px' />,
  url: FACEBOOK_URL
},]