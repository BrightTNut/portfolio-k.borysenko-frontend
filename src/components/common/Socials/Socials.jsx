import s from './Socials.module.css';
import { socials } from '../../../data/socials';

const Socials = () => {
  return (
    <div className={s.container}>
      <h2 className={s.title}>Follow Me</h2>
      <ul className={s.socials_list}>
        {socials.map(({ id, name, icon, url }) => (
          <a
            key={id}
            className={s.item}
            href={url}
            target="_blank"
            rel="noopener noreferrer"
          >
            {icon}
            <div className={s.name}>{name}</div>
          </a>
        ))}
      </ul>
    </div>
  );
};

export default Socials;
