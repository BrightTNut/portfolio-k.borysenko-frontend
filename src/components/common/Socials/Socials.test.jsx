import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import Socials from './Socials';
import { socials } from '../../../data/socials';

describe('Socials Component', () => {
  it('renders without crashing', () => {
    render(<Socials />);
    expect(screen.getByText('Follow Me')).toBeInTheDocument();
  });

  it('displays a list of social media links', () => {
    render(<Socials />);
    socials.forEach(social => {
      expect(screen.getByText(social.name)).toBeInTheDocument();
    });
  });

  it('has correct href for each social media link', () => {
    render(<Socials />);
    const links = screen.getAllByRole('link');

    links.forEach((link, index) => {
      expect(link).toHaveAttribute('href', socials[index].url);
    });
  });

  it('has target="_blank" and rel="noopener noreferrer" for each link', () => {
    render(<Socials />);
    socials.forEach(() => {
      const links = screen.getAllByRole('link');
      links.forEach(link => {
        expect(link).toHaveAttribute('target', '_blank');
        expect(link).toHaveAttribute('rel', 'noopener noreferrer');
      });
    });
  });
});
