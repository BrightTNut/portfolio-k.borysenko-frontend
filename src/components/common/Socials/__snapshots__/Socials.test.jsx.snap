// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Socials Component renders correctly and matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <h2
      class="title"
    >
      Follow Me
    </h2>
    <ul
      class="socials_list"
    >
      <a
        class="item"
        href="https://github.com/Kateryna-Borysenko"
        rel="noopener noreferrer"
        target="_blank"
      >
        <svg
          width="44px"
        >
          github.svg
        </svg>
        <div
          class="name"
        >
          GIT HUB
        </div>
      </a>
      <a
        class="item"
        href="https://www.linkedin.com/in/kateryna-borysenko/"
        rel="noopener noreferrer"
        target="_blank"
      >
        <svg
          width="44px"
        >
          linkedin.svg
        </svg>
        <div
          class="name"
        >
          LINKED IN
        </div>
      </a>
      <a
        class="item"
        href="https://www.instagram.com/k___borysenko/"
        rel="noopener noreferrer"
        target="_blank"
      >
        <svg
          width="44px"
        >
          instagram.svg
        </svg>
        <div
          class="name"
        >
          INSTAGRAM
        </div>
      </a>
      <a
        class="item"
        href="https://www.facebook.com/profile.php?id=100089711004988"
        rel="noopener noreferrer"
        target="_blank"
      >
        <svg
          width="44px"
        >
          facebook.svg
        </svg>
        <div
          class="name"
        >
          FACEBOOK
        </div>
      </a>
    </ul>
  </div>
</DocumentFragment>
`;
